src/managers/PlayerManager.js

```diff
@@ -1,20 +1,58 @@
 const Player = require('../structures/Player');
 const ReqUtil = require('../util/ReqUtil');
-
+const CacheManager = require('./CacheManager');
 class PlayerManager {
     constructor(client){
         this.client = client;
+
+        this.cache = new CacheManager(client, Player);
     }
 
     /**
      * Fetches a player and returns its data
      * @param {String} accountid The account ID or its tm.io vanity name
+     * @param {Boolean} cache Whether to get the player from cache or not
+     * @returns {Promise<Player>} The player
+     * @example
+     * // Get a player 
+     * client.players.get('26d9a7de-4067-4926-9d93-2fe62cd869fc').then(player => {
+     * console.log(player.displayname);
+     * });
+     */
+    async get(accountid, cache = this.client.options.cache.enabled){
+        if (cache) {
+            if (this.cache.has(accountid)) {
+                return this.cache.get(accountid);
+            } else {
+                return await this._fetch(accountid, cache);
+            }
+        } else {
+            return await this._fetch(accountid, cache);
+        }
+    }
+        
+    /**
+     * Fetches a player and returns its data
+     * @param {String} accountid The account ID or its tm.io vanity name
+     * @param {Boolean} cache Whether to cache the player or not
      * @returns {Player} The player
+     * @private
      */
-    async fetch(accountid){
+    async _fetch(accountid, cache = this.client.options.cache.enabled){
         const player = this.client.options.api.paths.tmio.tabs.player;
         const res = await this.client._apiReq(`${new ReqUtil(this.client).tmioAPIURL}/${player}/${accountid}`);
-        return new Player(this.client, res);
+        if (cache) {
+            res._cachedTimestamp = Date.now();
+            this.cache.set(res.accountid, new Player(this.client, res));
+
+            // Adds also the player by its vanity name in the cache
+            if (res.meta && (res.meta.vanity && res.meta.vanity != "")){
+                this.cache.set(res.meta.vanity, new Player(this.client, res));
+            }
+            return this.cache.get(res.accountid);
```

Comment:
There's not really any need to get this from the cache, right? We can just return the player value. Also, we should be able to only instantiate the Player once and reuse that, right?

Reply:
Fixed, we have now one instance of this Player